### PR TITLE
:book: Update document regarding overrides

### LINKED ISSUE #5818: clarify how clusterctl overrides works
**What steps did you take and what happened:**
[A clear and concise description on how to REPRODUCE the bug.]
`clusterctl upgrade plan | apply` ignores the overrides repository and fetch the metadata.yaml file from upstream.
Here is my overrides structure:
```sh
.cluster-api/
├── clusterctl.yaml
├── overrides
│   └── infrastructure-metal3
│       └── v1.0.0
└── version.yaml

```
clusterctl.yaml:
```yaml
KUBERNETES_VERSION: v1.22.2
NAMESPACE: metal3
NUM_OF_MASTER_REPLICAS: 3
NUM_OF_WORKER_REPLICAS: 1
NODE_DRAIN_TIMEOUT: 0s
MAX_SURGE_VALUE: 0

## Cluster network
CLUSTER_APIENDPOINT_HOST: 192.168.111.249
CLUSTER_APIENDPOINT_PORT: 6443
POD_CIDR: 192.168.0.0/18
SERVICE_CIDR: 10.96.0.0/12
PROVISIONING_POOL_RANGE_START: 172.22.0.100
PROVISIONING_POOL_RANGE_END: 172.22.0.200
PROVISIONING_CIDR: 24
BAREMETALV4_POOL_RANGE_START: 192.168.111.100
BAREMETALV4_POOL_RANGE_END: 192.168.111.200
BAREMETALV6_POOL_RANGE_START: fd55::100
BAREMETALV6_POOL_RANGE_END: fd55::200
EXTERNAL_SUBNET_V4_PREFIX: 24
EXTERNAL_SUBNET_V4_HOST: 192.168.111.1
EXTERNAL_SUBNET_V6_PREFIX: 64
EXTERNAL_SUBNET_V6_HOST: fd55::1
```
and this is the debug message from `clusterctl upgrade plan`:
```sh
$ clusterctl upgrade plan -v 9
Using configuration File="/home/ubuntu/.cluster-api/clusterctl.yaml"
Checking cert-manager version...
Cert-Manager is already up to date

Checking new release availability...
Fetching File="metadata.yaml" Provider="cluster-api" Type="CoreProvider" Version="v1.0.2"
Fetching File="metadata.yaml" Provider="kubeadm" Type="BootstrapProvider" Version="v1.0.2"
Fetching File="metadata.yaml" Provider="kubeadm" Type="ControlPlaneProvider" Version="v1.0.2"
Fetching File="metadata.yaml" Provider="cluster-api" Type="CoreProvider" Version="v1.0.2"
Fetching File="metadata.yaml" Provider="metal3" Type="InfrastructureProvider" Version="v0.5.3"
```
If the metal3 infrastructure provider metadata.yaml is used from the overrides directory, the last message should be:
`Using Override="metadata.yaml" Provider="infrastructure-metal3" Version="v1.0.0"`

**What did you expect to happen:**
When setting up the overrides directory for a provider, it is expected that clusterctl upgrade plan | apply would use the metadata.yaml file in this overrides directory instead of fetching it from upstream. This behavior is true with clusterctl v0.4.4. However, it is not true with >=v1.0.0.

**Environment:**

- Cluster-api version: >=v1.0.0
- Minikube/KIND version: kind version 0.11.1
- Kubernetes version: (use `kubectl version`):
Client: v1.22.4
Server: v1.22.2
- OS (e.g. from `/etc/os-release`): Ubuntu 20.04.3 LTS
/kind bug 
/area upgrade clusterctl
[One or more /area label. See https://github.com/kubernetes-sigs/cluster-api/labels?q=area for the list of labels]

### PR DESCRIPTION
**Pull request by Clash-ion**
**What this PR does / why we need it**:
Document the fact that overrides only provide file replacements; instead, provider version resolution is based only on the actual repository structure
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #5818

### CODE REVIEW

**Comment by Clash-ion** [comment]
are the above changes all right or they need some modification???

**Review by killianmuldoon — COMMENTED** [review]
/retitle :seedling: Update document regarding overrides

**Comment by Clash-ion** [comment]
> /retitle 🌱 Update document regarding overrides

what is the reason to put seedling emoji in title ?

**Comment by killianmuldoon** [comment]
We use the emojis for our release note automation - it gives us a clean way to divide up the 'kind' of PR. Thinking again I put the wrong one here :laughing: 

/retitle :book: Update document regarding overrides

**Comment by Clash-ion** [comment]
> We use the emojis for our release note automation - it gives us a clean way to divide up the 'kind' of PR. Thinking again I put the wrong one here 😆
> 
> /retitle 📖 Update document regarding overrides

ooo okay so that's what it is for...

**Comment by Clash-ion** [comment]
> /lgtm 
> 

hello, why did the pr check failed??

**Comment by sbueringer** [comment]
> > /lgtm
> 
> hello, why did the pr check failed??

Which one? I didn't see it before but we have a check that verify that the icon is in the PR title.

File: docs/book/src/clusterctl/configuration.md [hunk 1]
```diff
@@ -158,6 +158,7 @@ directory in the clusterctl config file as
 ```yaml
 overridesFolder: /Users/foobar/workspace/dev-releases
 ```
+Overrides only provide file replacements; instead, provider version resolution is based only on the actual repository structure.
```
Comment (fabriziopandini) [inline docs/book/src/clusterctl/configuration.md:161]: What about wrapping this note with 
```suggestion
<aside class="note warning">

<h1> Warning! </h1>

Overrides only provide file replacements; instead, provider version resolution is based only on the actual repository structure.

</aside>
```
to make it slightly more evident?

Comment (Clash-ion) [reply docs/book/src/clusterctl/configuration.md:161]: okay i ll update it .

File: docs/book/src/clusterctl/configuration.md [hunk 2]
```diff
@@ -158,6 +158,11 @@ directory in the clusterctl config file as
 ```yaml
 overridesFolder: /Users/foobar/workspace/dev-releases
 ```
+<aside class="note warning">
+<h1> Warning! </h1>
+Overrides only provide file replacements; instead, provider version resolution is based only on the actual repository structure.
+</aside>
+<br>
```
[the remark this answers is not in the document]
Comment (Clash-ion) [reply docs/book/src/clusterctl/configuration.md:165]: okay I ll do that

**Comment by fabriziopandini** [comment]
@Clash-ion any chance you can address the last nits?

**Comment by Clash-ion** [comment]
> @Clash-ion any chance you can address the last nit?

yes i ll try will try to update my pr today

**Comment by Clash-ion** [comment]
@fabriziopandini I think this looks good enough, I mean I just used the way it was previously done for Image Overrides.

**Pull request merged**